test/integration/gardenlet/networkpolicy/networkpolicy_test.go

```diff
@@ -0,0 +1,156 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy_test
+
+import (
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	networkpolicyhelper "github.com/gardener/gardener/pkg/gardenlet/controller/networkpolicy/helper"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Network policy controller tests", func() {
+
+	var (
+		expectedNetworkPolicySpec networkingv1.NetworkPolicySpec
+		shootNamespace            *corev1.Namespace
+		fooNamespace              *corev1.Namespace
+		kubernetesEndpoint        *corev1.Endpoints
+	)
+
+	BeforeEach(func() {
+		kubernetesEndpoint = &corev1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "kubernetes",
+			},
+		}
+
+		shootNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "shoot--",
+				Labels:       map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot},
+			},
+		}
+
+		fooNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "foo--",
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Construct expected network policy")
+		Expect(testClient.Get(ctx, client.ObjectKeyFromObject(kubernetesEndpoint), kubernetesEndpoint)).To(Succeed())
+
+		expectedNetworkPolicySpec = networkingv1.NetworkPolicySpec{
+			Egress: networkpolicyhelper.GetEgressRules(kubernetesEndpoint.Subsets...),
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"networking.gardener.cloud/to-seed-apiserver": "allowed"},
+			},
+			PolicyTypes: []networkingv1.PolicyType{"Egress"},
+		}
+
+		By("Create shoot namespace")
+		Expect(testClient.Create(ctx, shootNamespace)).To(Succeed())
+		log.Info("Created shoot namespace for test", "namespace", client.ObjectKeyFromObject(shootNamespace))
+
+		DeferCleanup(func() {
+			By("Delete shoot namespace")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shootNamespace))).To(Succeed())
+		})
+
+		By("Create foo namespace")
+		Expect(testClient.Create(ctx, fooNamespace)).To(Succeed())
+		log.Info("Created foo namespace for test", "namespace", client.ObjectKeyFromObject(fooNamespace))
+
+		DeferCleanup(func() {
+			By("Delete foo namespace")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, fooNamespace))).To(Succeed())
+		})
+	})
+
+	Context("#Reconcile", func() {
```

Comment:
```suggestion
	Context("reconciliation", func() {
```

We only use the `#...` notion for unit tests of a specific function.